protocols/v2/codec-sv2/src/encoder.rs

```diff
@@ -25,28 +69,61 @@ use buffer_sv2::{Buffer as IsBuffer, BufferFromSystemMemory as Buffer};
 #[cfg(feature = "with_buffer_pool")]
 use buffer_sv2::{Buffer as IsBuffer, BufferFromSystemMemory, BufferPool};
 
+/// The buffer type for holding intermediate data during encoding.
+///
+/// When the `with_buffer_pool` feature is enabled, `Buffer` is a pool-allocated buffer type
+/// (`BufferPool`), which allows for more efficient memory management. Otherwise, it defaults to
+/// `BufferFromSystemMemory`.
```

Comment:
```suggestion
/// When the `with_buffer_pool` feature is enabled, `Buffer` is a pool-allocated buffer type
/// (`BufferPool`), which allows for more efficient memory management. Otherwise, it defaults to
/// `binary_sv2::BufferFromSystemMemory`.
```

Reply:
we should reproduce this comment on `decoder.rs` because it also has a type alias for `Buffer`